tvm_utility/include/tvm_utility/pipeline.hpp

```diff
@@ -0,0 +1,282 @@
+/*
+ * Copyright 2020 Autoware Foundation. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <tvm_vendor/dlpack/dlpack.h>
+#include <string>
+#include <tvm_vendor/tvm/runtime/c_runtime_api.h>
+#include <tvm_vendor/tvm/runtime/module.h>
+#include <tvm_vendor/tvm/runtime/packed_func.h>
+#include <tvm_vendor/tvm/runtime/registry.h>
+#include <fstream>
+#include <vector>
+
+#pragma once
```

Comment:
Please use header guards instead of `#pragma once`. e.g.:

```
#ifndef TVM_UTILITY_PIPELINE_H
#define TVM_UTILITY_PIPELINE_H
```

Also, unfortunately, the standard for ROS1 header files is `.h` instead of `.hpp`. Please rename.

Reply:
Concerning the extension rename, `pipeline.hpp` is included in a file generated by TVM CLI. Since we want to have the same `tvm_utility` package in autoware.auto, I think we have to keep `.hpp` if we want this generated file to be usable in both autoware.ai and autoware.auto.

Reply:
That's fine. pre-generated files can be excluded from this rule. However, the other `include` files in this package which were hand-coded should be renamed to `.h` for the Autoware.AI version.

Reply:
I meant that in `inference_engine_tvm_config.hpp` (the pre-generated file) there is the following include: `#include "tvm_utility/pipeline.hpp"`. And since we want to have the same pre-generated file usable in the autoware.auto package (with the same `#include`), I think it prevents renaming `pipeline.hpp` (the hand-coded file) to `pipeline.h`. Otherwise we just move the problem to autoware.auto, where the standard is `.hpp` extension.

Reply:
Actually nevermind, TVM CLI will be modified to take care of that.
Done.